routers/web/web.go

```diff
@@ -672,16 +673,43 @@ func RegisterRoutes(m *web.Route) {
 			})
 			http.ServeFile(ctx.Resp, ctx.Req, path.Join(setting.StaticRootPath, "public/img/favicon.png"))
 		})
-		m.Group("/{username}", func() {
-			m.Get(".png", user.AvatarByUserName)
-			m.Get(".keys", user.ShowSSHKeys)
-			m.Get(".gpg", user.ShowGPGKeys)
-			m.Get(".rss", feedEnabled, feed.ShowUserFeedRSS)
-			m.Get(".atom", feedEnabled, feed.ShowUserFeedAtom)
-			m.Get("", user.Profile)
-		}, func(ctx *context.Context) {
-			ctx.Data["EnableFeed"] = setting.EnableFeed
-		}, context_service.UserAssignmentWeb())
+		m.Get("/{username}", func(ctx *context.Context) {
+			// WORKAROUND to support usernames with "." in it
```

Comment:
if somebody do have a better workaround please tell :/

Reply:
I guess this design would cause problem in the future.

For example, what if a user/org is called `server.ssh` or `mylibrary.png` ?

Should we introduce a breaking change to move these suffixes to a never-conflicting sub path?

Reply:
@wxiaoguang that's how it actually works atm!

this just fix an edgecase where the name is not "abc" but e.g. "ab.c" for example

Reply:
https://github.com/go-gitea/gitea/blob/main/models/user/user.go#L540

Reply:
But I see a ".png" there ....

Reply:
I also agree that we should not add more to that list ...

Reply:
@wxiaoguang well we just discovered an edgecase ... we could either drop one and break things or add it and might break another thing ...

Reply:
I would suggest to drop the `.png` here. Users could use `https://try.gitea.io/user/avatar/wxiaoguang/0` to fetch avatar image (well, this URL could also be fine-tuned). The `.png` seems quite misleading too, why not `.jpg` ? 😁 .

And maybe it's better to leave a big warning comment here to prevent future developers from adding more extensions.